From 65b437933eafd414c1d5c400124487b68e020eb3 Mon Sep 17 00:00:00 2001
From: Nick Chusid <nchusid@google.com>
Date: Mon, 27 Jan 2025 22:10:56 +0000
Subject: [PATCH] Fail fast in dng sdk if an invalid tag count is seen

Bug: 346797131
Test: TreeHugger
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:84b0c687ce847c80a2454b206c046da8241db41b)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:76a8892b5f72aba2c8fc9a74187bd378219b15ca)
Merged-In: I76da86b063ce4a0a329aca2a1309997cd62ac58d
Change-Id: I76da86b063ce4a0a329aca2a1309997cd62ac58d
---
 source/dng_ifd.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/source/dng_ifd.cpp b/source/dng_ifd.cpp
index 7f22065..bf3fb2c 100644
--- a/source/dng_ifd.cpp
+++ b/source/dng_ifd.cpp
@@ -351,7 +351,8 @@ bool dng_ifd::ParseTag (dng_stream &stream,
 			
 			CheckTagType (parentCode, tagCode, tagType, ttShort);
 			
-			CheckTagCount (parentCode, tagCode, tagCount, 1, 0x0FFFF);
+			if (!CheckTagCount (parentCode, tagCode, tagCount, 1, 0x0FFFF))
+				return false;
 			
 			#if qDNGValidate
 			
@@ -973,7 +974,8 @@ bool dng_ifd::ParseTag (dng_stream &stream,
 			
 			CheckTagType (parentCode, tagCode, tagType, ttShort);
 			
-			CheckTagCount (parentCode, tagCode, tagCount, 1, fSamplesPerPixel);
+			if (!CheckTagCount (parentCode, tagCode, tagCount, 1, fSamplesPerPixel))
+				return false;
 			
 			#if qDNGValidate
 			
@@ -1025,7 +1027,8 @@ bool dng_ifd::ParseTag (dng_stream &stream,
 			
 			CheckTagType (parentCode, tagCode, tagType, ttShort);
 			
-			CheckTagCount (parentCode, tagCode, tagCount, fSamplesPerPixel);
+			if (!CheckTagCount (parentCode, tagCode, tagCount, fSamplesPerPixel))
+				return false;
 			
 			#if qDNGValidate
 			
-- 
2.48.1.262.g85cc9f2d1e-goog

